flink-runtime/src/main/java/org/apache/flink/runtime/state/ChangelogTaskLocalStateStore.java

```diff
@@ -181,6 +226,16 @@ public void pruneCheckpoints(LongPredicate pruningChecker, boolean breakOnceChec
     @Override
     public CompletableFuture<Void> dispose() {
         deleteMaterialization(id -> true);
+        // delete all ChangelogStateHandle in taskowned directory.
+        discardExecutor.execute(
+                () ->
+                        syncDiscardFileForCollection(
```

Comment:
Yes, this will delete other job's states.
I moved `aid_xx/taskowned` folder to `aid_xxx/jid_xxx/taskowned` to avoid this.

> Why don't we put this logic in LocalStateRegistry - that would allow to avoid clashing and guessing the locations

Because the `taskowned` folder needs to be deleted when `dispose()`, and `LocalStateRegistry` can only delete the files in the folder, can't delete the folder itself.